Failure after AS 4.1 beta 1

Using 1.2. Might be addressed in your newest changes.

```
* What went wrong:
Could not determine the dependencies of task ':about:countReleaseDexMethods'.
> Cannot query the value of this property because it has no value available.

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.internal.tasks.TaskDependencyResolveException: Could not determine the dependencies of task ':about:countReleaseDexMethods'.
        at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.getDependencies(CachingTaskDependencyResolveContext.java:69)
        at org.gradle.execution.plan.TaskDependencyResolver.resolveDependenciesFor(TaskDependencyResolver.java:46)
        at org.gradle.execution.plan.LocalTaskNode.getDependencies(LocalTaskNode.java:161)
        at org.gradle.execution.plan.LocalTaskNode.resolveDependencies(LocalTaskNode.java:129)
        at org.gradle.execution.plan.DefaultExecutionPlan.doAddNodes(DefaultExecutionPlan.java:165)
        at org.gradle.execution.plan.DefaultExecutionPlan.addEntryTasks(DefaultExecutionPlan.java:135)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph.addEntryTasks(DefaultTaskExecutionGraph.java:144)
        at org.gradle.execution.TaskNameResolvingBuildConfigurationAction.configure(TaskNameResolvingBuildConfigurationAction.java:49)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter.configure(DefaultBuildConfigurationActionExecuter.java:58)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter.access$200(DefaultBuildConfigurationActionExecuter.java:26)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter$2.proceed(DefaultBuildConfigurationActionExecuter.java:66)
        at org.gradle.execution.DefaultTasksBuildExecutionAction.configure(DefaultTasksBuildExecutionAction.java:45)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter.configure(DefaultBuildConfigurationActionExecuter.java:58)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter.access$200(DefaultBuildConfigurationActionExecuter.java:26)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter$2.proceed(DefaultBuildConfigurationActionExecuter.java:66)
        at org.gradle.execution.ExcludedTaskFilteringBuildConfigurationAction.configure(ExcludedTaskFilteringBuildConfigurationAction.java:48)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter.configure(DefaultBuildConfigurationActionExecuter.java:58)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter.access$200(DefaultBuildConfigurationActionExecuter.java:26)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter$1.run(DefaultBuildConfigurationActionExecuter.java:44)
        at org.gradle.internal.Factories$1.create(Factories.java:26)
        at org.gradle.api.internal.project.DefaultProjectStateRegistry.withLenientState(DefaultProjectStateRegistry.java:134)
        at org.gradle.api.internal.project.DefaultProjectStateRegistry.withLenientState(DefaultProjectStateRegistry.java:126)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter.select(DefaultBuildConfigurationActionExecuter.java:40)
        at org.gradle.initialization.DefaultTaskExecutionPreparer.prepareForTaskExecution(DefaultTaskExecutionPreparer.java:38)
        at org.gradle.initialization.BuildOperatingFiringTaskExecutionPreparer$CalculateTaskGraph.populateTaskGraph(BuildOperatingFiringTaskExecutionPreparer.java:117)
        at org.gradle.initialization.BuildOperatingFiringTaskExecutionPreparer$CalculateTaskGraph.run(BuildOperatingFiringTaskExecutionPreparer.java:68)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:395)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:387)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:157)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:242)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:150)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:84)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.initialization.BuildOperatingFiringTaskExecutionPreparer.prepareForTaskExecution(BuildOperatingFiringTaskExecutionPreparer.java:56)
        at org.gradle.initialization.DefaultGradleLauncher.prepareTaskExecution(DefaultGradleLauncher.java:227)
        at org.gradle.initialization.DefaultGradleLauncher.doClassicBuildStages(DefaultGradleLauncher.java:159)
        at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:140)
        at org.gradle.initialization.DefaultGradleLauncher.executeTasks(DefaultGradleLauncher.java:120)
        at org.gradle.internal.invocation.GradleBuildController$1.create(GradleBuildController.java:74)
        at org.gradle.internal.invocation.GradleBuildController$1.create(GradleBuildController.java:67)
        at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:180)
        at org.gradle.internal.work.StopShieldingWorkerLeaseService.withLocks(StopShieldingWorkerLeaseService.java:40)
        at org.gradle.internal.invocation.GradleBuildController.doBuild(GradleBuildController.java:67)
        at org.gradle.internal.invocation.GradleBuildController.run(GradleBuildController.java:56)
        at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:31)
        at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
        at org.gradle.launcher.exec.BuildOutcomeReportingBuildActionRunner.run(BuildOutcomeReportingBuildActionRunner.java:63)
        at org.gradle.tooling.internal.provider.ValidatingBuildActionRunner.run(ValidatingBuildActionRunner.java:32)
        at org.gradle.launcher.exec.BuildCompletionNotifyingBuildActionRunner.run(BuildCompletionNotifyingBuildActionRunner.java:39)
        at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner$3.call(RunAsBuildOperationBuildActionRunner.java:51)
        at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner$3.call(RunAsBuildOperationBuildActionRunner.java:45)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:409)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:399)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:157)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:242)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:150)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:94)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
        at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner.run(RunAsBuildOperationBuildActionRunner.java:45)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter$1.transform(InProcessBuildActionExecuter.java:50)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter$1.transform(InProcessBuildActionExecuter.java:47)
        at org.gradle.composite.internal.DefaultRootBuildState.run(DefaultRootBuildState.java:80)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:47)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:31)
        at org.gradle.launcher.exec.BuildTreeScopeBuildActionExecuter.execute(BuildTreeScopeBuildActionExecuter.java:42)
        at org.gradle.launcher.exec.BuildTreeScopeBuildActionExecuter.execute(BuildTreeScopeBuildActionExecuter.java:28)
        at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:87)
        at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:55)
        at org.gradle.tooling.internal.provider.SubscribableBuildActionExecuter.execute(SubscribableBuildActionExecuter.java:60)
        at org.gradle.tooling.internal.provider.SubscribableBuildActionExecuter.execute(SubscribableBuildActionExecuter.java:38)
        at org.gradle.tooling.internal.provider.SessionScopeBuildActionExecuter.execute(SessionScopeBuildActionExecuter.java:68)
        at org.gradle.tooling.internal.provider.SessionScopeBuildActionExecuter.execute(SessionScopeBuildActionExecuter.java:38)
        at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:37)
        at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:26)
        at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:60)
        at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:32)
        at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:56)
        at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:42)
        at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:48)
        at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:32)
        at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:68)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:39)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:29)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:35)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.create(ForwardClientInput.java:78)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.create(ForwardClientInput.java:75)
        at org.gradle.util.Swapper.swap(Swapper.java:38)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:75)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.LogAndCheckHealth.execute(LogAndCheckHealth.java:55)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:63)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:82)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:52)
        at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:297)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
Caused by: org.gradle.api.internal.provider.MissingValueException: Cannot query the value of this property because it has no value available.
        at org.gradle.api.internal.provider.AbstractMinimalProvider.get(AbstractMinimalProvider.java:86)
        at org.gradle.api.internal.file.collections.UnpackingVisitor.add(UnpackingVisitor.java:56)
        at org.gradle.api.internal.file.collections.DefaultConfigurableFileCollection$UnresolvedItemsCollector.visitContents(DefaultConfigurableFileCollection.java:300)
        at org.gradle.api.internal.file.collections.DefaultConfigurableFileCollection.visitContents(DefaultConfigurableFileCollection.java:224)
        at org.gradle.api.internal.file.CompositeFileCollection.visitDependencies(CompositeFileCollection.java:140)
        at org.gradle.api.internal.file.collections.DefaultConfigurableFileCollection.visitDependencies(DefaultConfigurableFileCollection.java:230)
        at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext$TaskGraphImpl.getNodeValues(CachingTaskDependencyResolveContext.java:112)
        at org.gradle.internal.graph.CachingDirectedGraphWalker$GraphWithEmptyEdges.getNodeValues(CachingDirectedGraphWalker.java:213)
        at org.gradle.internal.graph.CachingDirectedGraphWalker.doSearch(CachingDirectedGraphWalker.java:121)
        at org.gradle.internal.graph.CachingDirectedGraphWalker.findValues(CachingDirectedGraphWalker.java:73)
        at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.getDependencies(CachingTaskDependencyResolveContext.java:67)
        ... 106 more

```

Trying to sync gradle via the IDE just says `Extension not initialized yet, couldn't access compileSdkVersion.
`, which is probably a generic message for sync errors. The problem can be replicated solely by updating AS, and using the new plugin + new gradle

--- a/.buildscript/test.sh
+++ b/.buildscript/test.sh
@@ -23,29 +23,29 @@ function die() {
   exit 1
 }
 
-grep -F 'Total methods in app-debug-it.apk: 7355 (11.22% used)' app.log || die "Incorrect method count in app-debug-it.apk"
-grep -F 'Total fields in app-debug-it.apk:  3780 (5.77% used)' app.log || die "Incorrect field count in app-debug-it.apk"
-grep -F 'Total classes in app-debug-it.apk:  926 (1.41% used)' app.log || die "Incorrect field count in app-debug-it.apk"
-grep -F 'Methods remaining in app-debug-it.apk: 58180' app.log || die "Incorrect remaining-method value in app-debug-it.apk"
-grep -F 'Fields remaining in app-debug-it.apk:  61755' app.log || die "Incorrect remaining-field value in app-debug-it.apk"
-grep -F 'Classes remaining in app-debug-it.apk:  64609' app.log || die "Incorrect remaining-field value in app-debug-it.apk"
+grep -F 'Total methods in app-debug-it.apk: 7356 (11.22% used)' app.log || die "Incorrect method count in app-debug-it.apk"
+grep -F 'Total fields in app-debug-it.apk:  2597 (3.96% used)' app.log || die "Incorrect field count in app-debug-it.apk"
+grep -F 'Total classes in app-debug-it.apk:  441 (0.67% used)' app.log || die "Incorrect field count in app-debug-it.apk"
+grep -F 'Methods remaining in app-debug-it.apk: 58179' app.log || die "Incorrect remaining-method value in app-debug-it.apk"
+grep -F 'Fields remaining in app-debug-it.apk:  62938' app.log || die "Incorrect remaining-field value in app-debug-it.apk"
+grep -F 'Classes remaining in app-debug-it.apk:  65094' app.log || die "Incorrect remaining-field value in app-debug-it.apk"
 
-grep -F "##teamcity[buildStatisticValue key='Dexcount_app_debug_ClassCount' value='926']" app.log || die "Missing or incorrect Teamcity method count value"
-grep -F "##teamcity[buildStatisticValue key='Dexcount_app_debug_MethodCount' value='7355']" app.log || die "Missing or incorrect Teamcity method count value"
-grep -F "##teamcity[buildStatisticValue key='Dexcount_app_debug_FieldCount' value='3780']" app.log || die "Missing or incorrect Teamcity field count value"
+grep -F "##teamcity[buildStatisticValue key='Dexcount_app_debug_ClassCount' value='441']" app.log || die "Missing or incorrect Teamcity method count value"
+grep -F "##teamcity[buildStatisticValue key='Dexcount_app_debug_MethodCount' value='7356']" app.log || die "Missing or incorrect Teamcity method count value"
+grep -F "##teamcity[buildStatisticValue key='Dexcount_app_debug_FieldCount' value='2597']" app.log || die "Missing or incorrect Teamcity field count value"
 
-grep -F 'Total methods in tests-debug.apk: 4265 (6.51% used)' tests.log || die "Incorrect method count in tests-debug.apk"
-grep -F 'Total fields in tests-debug.apk:  1270 (1.94% used)' tests.log || die "Incorrect field count in tests-debug.apk"
+grep -F 'Total methods in tests-debug.apk: 4266 (6.51% used)' tests.log || die "Incorrect method count in tests-debug.apk"
+grep -F 'Total fields in tests-debug.apk:  1268 (1.93% used)' tests.log || die "Incorrect field count in tests-debug.apk"
 grep -F 'Total classes in tests-debug.apk:  723 (1.10% used)' tests.log || die "Incorrect field count in tests-debug.apk"
-grep -F 'Methods remaining in tests-debug.apk: 61270' tests.log || die "Incorrect remaining-method value in tests-debug.apk"
-grep -F 'Fields remaining in tests-debug.apk:  64265' tests.log || die "Incorrect remaining-field value in tests-debug.apk"
+grep -F 'Methods remaining in tests-debug.apk: 61269' tests.log || die "Incorrect remaining-method value in tests-debug.apk"
+grep -F 'Fields remaining in tests-debug.apk:  64267' tests.log || die "Incorrect remaining-field value in tests-debug.apk"
 grep -F 'Classes remaining in tests-debug.apk:  64812' tests.log || die "Incorrect remaining-field value in tests-debug.apk"
 
 grep -F 'Total methods in lib-debug.aar: 7 (0.01% used)' lib.log || die "Incorrect method count in lib-debug.aar"
-grep -F 'Total fields in lib-debug.aar:  5 (0.01% used)' lib.log || die "Incorrect field count in lib-debug.aar"
+grep -F 'Total fields in lib-debug.aar:  3 (0.00% used)' lib.log || die "Incorrect field count in lib-debug.aar"
 grep -F 'Total classes in lib-debug.aar:  5 (0.01% used)' lib.log || die "Incorrect class count in lib-debug.aar"
 grep -F 'Methods remaining in lib-debug.aar: 65528' lib.log || die "Incorrect remaining-method count in lib-debug.aar"
-grep -F 'Fields remaining in lib-debug.aar:  65530' lib.log || die "Incorrect remaining-field count in lib-debug.aar"
+grep -F 'Fields remaining in lib-debug.aar:  65532' lib.log || die "Incorrect remaining-field count in lib-debug.aar"
 grep -F 'Classes remaining in lib-debug.aar:  65530' lib.log || die "Incorrect remaining-class count in lib-debug.aar"
 
 # Note the '&&' here - grep exits with an error if no lines match,

--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -28,7 +28,7 @@ ext {
 ext.deps = [
     // Plugins
     "gradle"              : [
-        dependencies.create("com.android.tools.build:gradle:4.0.0") {
+        dependencies.create("com.android.tools.build:gradle:4.1.0-beta01") {
             // Android build tools (as of 3.2.0-alpha18) bundle the deprecated
             // 'jre' stdlib modules, which cause warnings at build-time that
             // fail the build.

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.4-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-all.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists

--- a/src/main/kotlin/com/getkeepsafe/dexcount/DexCountTask.kt
+++ b/src/main/kotlin/com/getkeepsafe/dexcount/DexCountTask.kt
@@ -18,8 +18,8 @@ package com.getkeepsafe.dexcount
 
 import org.gradle.api.DefaultTask
 import org.gradle.api.GradleException
-import org.gradle.api.file.ConfigurableFileCollection
 import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.FileCollection
 import org.gradle.api.file.RegularFileProperty
 import org.gradle.api.logging.LogLevel
 import org.gradle.api.model.ObjectFactory
@@ -65,7 +65,7 @@ open class DexCountTask @Inject constructor(
 
     @Nullable
     @InputFiles
-    val mappingFileProvider: ConfigurableFileCollection = objects.fileCollection()
+    val mappingFileProvider: Property<FileCollection> = objects.property(FileCollection::class.java)
 
     @OutputFile
     val outputFile: RegularFileProperty = objects.fileProperty()
@@ -101,17 +101,15 @@ open class DexCountTask @Inject constructor(
     }
 
     private val deobfuscator by lazy {
-        val fileCollection = mappingFileProvider
-        val file = if (fileCollection.isEmpty) null else fileCollection.singleFile
-        when {
-            file == null -> Deobfuscator.empty
-            !file.exists() -> {
-                withStyledOutput(level = LogLevel.DEBUG) {
-                    it.println("Mapping file specified at ${file.absolutePath} does not exist, assuming output is not obfuscated.")
-                }
-                Deobfuscator.empty
+        val fileCollection = mappingFileProvider.orNull ?: return@lazy Deobfuscator.empty
+        val file = fileCollection.singleOrNull() ?: return@lazy Deobfuscator.empty
+        if (file.exists()) {
+            Deobfuscator.create(file)
+        } else {
+            withStyledOutput(level = LogLevel.DEBUG) {
+                it.println("Mapping file specified at ${file.absolutePath} does not exist, assuming output is not obfuscated.")
             }
-            else -> Deobfuscator.create(file)
+            Deobfuscator.empty
         }
     }
 

--- a/src/main/kotlin/com/getkeepsafe/dexcount/Plugin.kt
+++ b/src/main/kotlin/com/getkeepsafe/dexcount/Plugin.kt
@@ -52,6 +52,7 @@ open class DexMethodCountPlugin : Plugin<Project> {
         private const val AGP_VERSION_3: String = "3.0.0"
         private const val AGP_VERSION_3_3 = "3.3.0"
         private const val AGP_VERSION_3_6 = "3.6.0"
+        private const val AGP_VERSION_4_1 = "4.1.0"
         private const val ANDROID_EXTENSION_NAME = "android"
         private const val SDK_DIRECTORY_METHOD = "getSdkDirectory"
 
@@ -89,12 +90,21 @@ open class DexMethodCountPlugin : Plugin<Project> {
             throw IllegalStateException("dexcount requires the Android plugin to be configured")
         }
 
-        val android = project.extensions.findByName(ANDROID_EXTENSION_NAME)
-        sdkLocation = android?.javaClass?.getMethod(SDK_DIRECTORY_METHOD)?.invoke(android) as File?
+        // It's important not to invoke the private 'getSdkDirectory()' method until
+        // *after* the project is evaluated; if we access it too early (as of 4.1.0-beta01)
+        // we get a rather obscure exception about 'compileSdkVersion' not being set because
+        // the extension is not yet initialized.
+        project.afterEvaluate {
+            val android = it.extensions.findByName(ANDROID_EXTENSION_NAME)
+            val extClass = android?.javaClass
+            val getSdkDirectory = extClass?.getMethod(SDK_DIRECTORY_METHOD)
+            sdkLocation = getSdkDirectory?.invoke(android) as File?
+        }
 
         val gradlePluginRevision = Revision.parseRevision(gradlePluginVersion, Revision.Precision.PREVIEW)
 
         val applicator = when {
+            gradlePluginRevision isAtLeast AGP_VERSION_4_1 -> FourOneApplicator(project)
             gradlePluginRevision isAtLeast AGP_VERSION_3_6 -> ThreeSixApplicator(project)
             gradlePluginRevision isAtLeast AGP_VERSION_3_3 -> ThreeThreeApplicator(project)
             gradlePluginRevision isAtLeast AGP_VERSION_3 -> ThreeOhApplicator(project)
@@ -225,7 +235,7 @@ abstract class TaskApplicator(
             t.config              = ext
 
             t.variantOutputName.set(outputName)
-            t.mappingFileProvider.from(getMappingFile(variant))
+            t.mappingFileProvider.set(getMappingFile(variant))
             t.outputFile.set(project.file(path + (ext.format as OutputFormat).extension))
             t.summaryFile.set(project.file(path + ".csv"))
             t.chartDir.set(project.file(path + "Chart"))
@@ -248,7 +258,7 @@ abstract class TaskApplicator(
                 description = "Outputs declared method count."
                 group = "Reporting"
                 variantOutputName.set("")
-                mappingFileProvider.from(project.files())
+                mappingFileProvider.set(project.files())
                 outputFile.set(File(outputDir, name + (ext.format as OutputFormat).extension))
                 summaryFile.set(File(outputDir, "$name.csv"))
                 chartDir.set(File(outputDir, name + "Chart"))
@@ -384,12 +394,21 @@ open class ThreeThreeApplicator(project: Project): TaskApplicator(project) {
     }
 }
 
-class ThreeSixApplicator(project: Project) : ThreeThreeApplicator(project) {
+open class ThreeSixApplicator(project: Project) : ThreeThreeApplicator(project) {
     override fun getOutputDirectory(task: PackageAndroidArtifact): DirectoryProperty {
         return task.outputDirectory
     }
 
     override fun getMappingFile(variant: BaseVariant): Provider<FileCollection> {
         return variant.mappingFileProvider
+    }
+}
+
+open class FourOneApplicator(project: Project) : ThreeSixApplicator(project) {
+    override fun getMappingFile(variant: BaseVariant): Provider<FileCollection> {
+        return project.provider {
+            project.logger.quiet("NOTE: Deobfuscation is not supported in AGP 4.1.0 (see issue https://issuetracker.google.com/u/1/issues/158379522)")
+            project.files()
+        }
     }
 }

--- a/src/test/groovy/com/getkeepsafe/dexcount/DexMethodCountPluginSpec.groovy
+++ b/src/test/groovy/com/getkeepsafe/dexcount/DexMethodCountPluginSpec.groovy
@@ -98,7 +98,7 @@ final class DexMethodCountPluginSpec extends Specification {
           compileSdkVersion 28
 
           defaultConfig {
-            if ("${projectPlugin}" == 'com.android.application') { 
+            if ("${projectPlugin}" == 'com.android.application') {
               applicationId 'com.example'
             }
           }
@@ -203,7 +203,7 @@ final class DexMethodCountPluginSpec extends Specification {
             classpath files($classpathString)
           }
         }
-        
+
         // TODO(???) - Repositories from test
         repositories {
           google()
@@ -225,7 +225,7 @@ final class DexMethodCountPluginSpec extends Specification {
         when:
         def result = GradleRunner.create()
             .withProjectDir(testProjectDir.root)
-            .withArguments("${taskName}")
+            .withArguments("${taskName}", "--full-stacktrace")
             .build()
 
         then:
@@ -249,7 +249,7 @@ final class DexMethodCountPluginSpec extends Specification {
             classpath files($classpathString)
           }
         }
-        
+
         // TODO(???) - Repositories from test
         repositories {
           google()
@@ -265,7 +265,7 @@ final class DexMethodCountPluginSpec extends Specification {
           defaultConfig {
             applicationId 'com.example'
           }
-          
+
           buildTypes {
            debug {}
            release {}
@@ -300,7 +300,7 @@ final class DexMethodCountPluginSpec extends Specification {
             classpath files($classpathString)
           }
         }
-        
+
         // TODO(???) - Repositories from test
         repositories {
           google()
@@ -316,14 +316,14 @@ final class DexMethodCountPluginSpec extends Specification {
           defaultConfig {
             applicationId 'com.example'
           }
-          
+
           buildTypes {
             debug {}
             release {}
           }
-    
+
           flavorDimensions 'a', 'b'
-    
+
           productFlavors {
             flavor1 { dimension 'a' }
             flavor2 { dimension 'a' }
@@ -367,7 +367,7 @@ final class DexMethodCountPluginSpec extends Specification {
             classpath files($classpathString)
           }
         }
-        
+
         // TODO(???) - Repositories from test
         repositories {
           google()
@@ -424,7 +424,7 @@ final class DexMethodCountPluginSpec extends Specification {
             applicationId 'com.example'
           }
         }
-        
+
         dexcount {
           enabled = false
         }

